redhat-cop/automate_tower_genie_collections
# feat: support of export single project/workflow/job

# What does this PR do?
The tasks should allow users to easily export all objects related to a specific project.

Currently, filetree_create retrieves every project and other defined objects, which is not desirable in the following use case scenario.

Use case scenario example for a project: 

1. The user creates a project in the QA AAP environment.
2. The user create many of job templates that use this project
3. The user wants to promote the project with job templates to the Production environment.
4. The user uses filetree_create to create a copy of the project with job templates and push the files to a remote repository.
5. The user uses filetree_read to import the object from the repository.

# How should this be tested?
Manual
1) Provide the variables to the usage of the role:
```
- name: Get project
  hosts: all 
  roles:
    - role: infra.controller_configuration.filetree_create
      vars:
        project_id: 4
  ```      
2) Execute the playbook and check if the only relevant data were exported.

# Is there a relevant Issue open for this?
N/A

# Other Relevant info, PRs, etc
Request on the forum, that what we want to achieve: https://forum.ansible.com/t/rfe-develop-new-functionality-for-exporting-configurations-filtered-by-project/5220
